### PR TITLE
add templates for realgeeks website/crm service

### PR DESCRIPTION
**Pull request by slyboots**
Templates for Real Geeks web platform

### CODE REVIEW

File: realgeeks.com.website_and_crm.json [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      "pointsTo": "mxb.mailgun.org",
+      "priority": 10,
+      "ttl": 300
+    },
+    {
+      "groupId": "CRM",
+      "type": "SPFM",
+      "host": "rg-mail.www",
+      "spfRules": "v=spf1 include:mailgun.org ~all"
+    },
```
Comment (arnoldblinn) [inline realgeeks.com.website_and_crm.json:46]: With SPFM, you don't want the v=spf1 or ~all. See 6.10.3 in the spec

File: realgeeks.com.website_and_crm.json [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  "description": "Enables a domain to be used for a Real Geeks website and CRM",
+  "variableDescription": "Self explanatory in the variable names.",
+  "records": [
+    {
+      "groupId": "Website",
+      "type": "APEXCNAME",
+      "host": "@",
+      "pointsTo": "%PRODHOST%",
+      "ttl": 600
+    },
```
Comment (arnoldblinn) [inline realgeeks.com.website_and_crm.json:17]: APEXCNAME will severely limit you on DNS Providers you can work with; but this is your perogative.

Also recommend that you make variable names lower case....experience tells me this is a bit more friendly.

You also are going to want to digitally sign this template.  A synchronous template with a variable in pointsTo is susceptible to fishing attacks.  See section 4.2.3 of the spec

File: realgeeks.com.website_and_crm.json (hunk 1 above)
Comment (slyboots) [reply realgeeks.com.website_and_crm.json:46]: Ahh, thank you!

File: realgeeks.com.website_and_crm.json (hunk 2 above)
Comment (slyboots) [reply realgeeks.com.website_and_crm.json:17]: The APEXCNAME is actually one of the requirements we had for using Domain Connect. Luckily GoDaddy supports it, and they're currently the only ones we're wanting to work with.

The caps lock variables felt bad to type honestly.

This is meant to be a synchronous template, so it definitely needs to be signed. Thanks for the input!

Comment (slyboots) [reply realgeeks.com.website_and_crm.json:18]: Added the syncPubKeyDomain key to the template.

**Comment by arnoldblinn** [comment]
We had to update the build (which is a schema check) to make your pull request pass our Travis CI build.  The schema check didn't like the APEXCNAME (which you are really the first company to depend on....like I said use of this fields severely limits which DNS Providers you can work with). Please merge master back and then I can merge your pull request.

**Comment by slyboots** [comment]
I've merged master back in. It seems like the build is still failing though...
I didnt realize what we were wanting was so far outside of the norm since it was mentioned in the specification.
I really do appreciate your help despite the extra work Im causing 😅

**Comment by arnoldblinn** [comment]
https://travis-ci.com/github/Domain-Connect/Templates/builds/177478977

The error is that your serviceId doesn't match the name of the file.  This is required.  Either change your service id, or change the file name.

**Comment by pawel-kow** [comment]
> I've merged master back in. It seems like the build is still failing though...

This  time it's file naming issue

`realgeeks.com.website_and_crm.json not matching the pattern. Expected realgeeks.com.realgeeks.json`

**Comment by pawel-kow** [comment]
Btw... well APEXCNAME is something barely anyone supports. For a reason it's in extension part odd the spec.  Main problem is it does not work with pre signed zones with DNSSEC.
I understand you want to work with GD but in fact investing in Domain Connect it's a pity and a bit short sighted AFAIK as other providers would work out of the box.

**Comment by slyboots** [comment]
Ahh, I didnt realize I could actually see the build results until just now. I should have clicked around a bit more. Thank you both.

>I understand you want to work with GD but in fact investing in Domain Connect it's a pity and a bit short sighted AFAIK as other providers would work out of the box.
I dont disagree with you, but ~80% of our client base uses godaddy. We cant really ask them to switch to a different registrar. Changing our infrastructure so that we dont require an APEXCNAME is easier said than done though. So If we have an opportunity to improve our support for the majority of our clients with minimal changes right now it seems reasonable to take advantage of it and then focus on further improving how we use Domain Connect.

**Pull request merged**